ccm_web/server/app.ts

```diff
@@ -0,0 +1,61 @@
+import path from 'path'
+
+import express from 'express'
+import webpack from 'webpack'
+import webpackDevMiddleware from 'webpack-dev-middleware'
+
+import devConfig from '../webpack/webpack.dev'
+
+
+const compiler = webpack(devConfig)
+const { NODE_ENV, PORT } = process.env
+
+const port = PORT || 4000
+const isDev = NODE_ENV !== 'production'
+
+
+// Initialize
+const app = express()
+
+// API Routes
+
+const apiPathBase = '/api/'
+
+// Added for proof of concept
+app.get(apiPathBase + 'hello', (req, res) => {
+  let message = `Status code: ${res.statusCode}.`
+  if (res.statusCode === 200) {
+    message += ' You successfully communicated with the backend server. Hooray!'
+  }
+  res.json({ message: message })
+})
+
+// Handle client code
+if (isDev) {
+  console.log('Setting up webpack-dev-middleware...')
+  const publicPath = typeof devConfig.output?.publicPath === 'string'
+    ? devConfig.output.publicPath : undefined
+
+  app.use(
+    webpackDevMiddleware(compiler, {
+      publicPath: publicPath,
+      // Just copies what's used in memory to client/build (changing doesn't do anything)
+      writeToDisk: true
+    })
+  )
+
```

Comment:
This works out pretty nicely in development, but it's a little awkward having to import `webpack` and related dependencies even in the production images. Now or later, we might want to think about either having a separate `dev` `app.ts`. Alternatively, we could use `webpack-dev-server`, but that starts to look similar to `react-scripts`. There might be a way to run two servers in one container? I haven't thought that through too much.

Reply:
See issue #29.